00_Base/src/services/credentials.service.ts

```diff
@@ -474,6 +477,119 @@ export class CredentialsService {
     return updatedClientInformation;
   }
 
+  async generateCredentialsTokenA(
+    credentialsRequest: CredentialsRequestDTO,
+    versionNumber: VersionNumber,
+  ): Promise<CredentialsDTO> {
+    const serverCredentialsRoleDTOs = credentialsRequest.roles;
+
+    for (const role of serverCredentialsRoleDTOs) {
```

Comment:
Maybe consider wrapping this for loop in a method `validateServerRolesForRegistration`?

I would also think that the logic for `registerCredentialsTokenA` should also have this same validation logic, no?

Reply:
`registerCredentialsTokenA` doesn't has the save validation in our current code and we assume we should always create new objects since the register process happens at the very beginning. Also based on the OCPI protocol, when registering, receiver (MSP) only send us version url and token A. So we should not need roles. We might refactor the request body of the register API.